diff -ru grandorgue-3.14.1.orig/src/core/GOStdPath.cpp grandorgue-3.14.1/src/core/GOStdPath.cpp
--- grandorgue-3.14.1.orig/src/core/GOStdPath.cpp	2024-04-26 15:16:05.052857222 -0300
+++ grandorgue-3.14.1/src/core/GOStdPath.cpp	2024-04-26 15:16:39.842830713 -0300
@@ -41,6 +41,14 @@
   return wxStandardPaths::Get().GetUserConfigDir();
 }
 
+wxString GOStdPath::GetCacheDir() {
+#if wxCHECK_VERSION(3, 1, 1)
+    return wxStandardPaths::Get().GetUserDir(wxStandardPaths::Dir_Cache);
+#else
+    return GetGrandOrgueSubDir(wxT("Cache"));
+#endif
+}
+
 wxString GOStdPath::GetDocumentDir() {
   return wxStandardPaths::Get().GetDocumentsDir();
 }
diff -ru grandorgue-3.14.1.orig/src/core/GOStdPath.h grandorgue-3.14.1/src/core/GOStdPath.h
--- grandorgue-3.14.1.orig/src/core/GOStdPath.h	2024-04-26 15:19:32.162697640 -0300
+++ grandorgue-3.14.1/src/core/GOStdPath.h	2024-04-26 15:18:15.472755558 -0300
@@ -16,6 +16,7 @@
   static wxString GetBaseDir();
   static wxString GetResourceDir();
   static wxString GetConfigDir();
+  static wxString GetCacheDir();
   static wxString GetDocumentDir();
   static wxString GetGrandOrgueDir();
   static wxString GetGrandOrgueSubDir(const wxString &subdir);
diff -ru grandorgue-3.14.1.orig/src/grandorgue/GOApp.cpp grandorgue-3.14.1/src/grandorgue/GOApp.cpp
--- grandorgue-3.14.1.orig/src/grandorgue/GOApp.cpp	2024-04-26 15:19:47.802685569 -0300
+++ grandorgue-3.14.1/src/grandorgue/GOApp.cpp	2024-04-26 15:20:49.972641605 -0300
@@ -147,10 +147,12 @@
   if (!wxApp::OnInit())
     return false;
 
+  // calls SetFileLayout
+  GOStdPath::InitLocaleDir();
+
   m_config = new GOConfig(m_InstanceName);
   m_config->Load();
 
-  GOStdPath::InitLocaleDir();
   m_locale.Init(m_config->GetLanguageId());
   m_locale.AddCatalog(wxT("GrandOrgue"));
 
diff -ru grandorgue-3.14.1.orig/src/grandorgue/config/GOConfig.cpp grandorgue-3.14.1/src/grandorgue/config/GOConfig.cpp
--- grandorgue-3.14.1.orig/src/grandorgue/config/GOConfig.cpp	2024-04-26 15:21:56.652590461 -0300
+++ grandorgue-3.14.1/src/grandorgue/config/GOConfig.cpp	2024-04-26 15:24:27.672471786 -0300
@@ -228,9 +228,9 @@
   OrganPackagePath.setDefaultValue(
     GOStdPath::GetGrandOrgueSubDir(_("Organ packages")));
   OrganCachePath.setDefaultValue(
-    GOStdPath::GetGrandOrgueSubDir(wxT("Cache") + m_InstanceName));
+    GOStdPath::GetCacheDir() + wxFileName::GetPathSeparator() + wxT("GrandOrgueCache") + m_InstanceName);
   OrganSettingsPath.setDefaultValue(
-    GOStdPath::GetGrandOrgueSubDir(wxT("Data") + m_InstanceName));
+    GOStdPath::GetConfigDir() + wxFileName::GetPathSeparator() + wxT("GrandOrgueData") + m_InstanceName);
   OrganCombinationsPath.setDefaultValue(
     GOStdPath::GetGrandOrgueSubDir(_("Combinations")));
   ExportImportPath.setDefaultValue(
